templates/types/streaming/nextjs/app/components/ui/file-uploader.tsx

```diff
@@ -50,30 +52,51 @@ export default function FileUploader({
   };
 
   const onFileChange = async (e: ChangeEvent<HTMLInputElement>) => {
-    const file = e.target.files?.[0];
-    if (!file) return;
+    const files = Array.from(e.target.files || []);
+    if (!files.length) return;
 
     setUploading(true);
-    await handleUpload(file);
+
+    await handleUpload(files);
+
     resetInput();
     setUploading(false);
   };
 
-  const handleUpload = async (file: File) => {
+  const handleUpload = async (files: File[]) => {
     const onFileUploadError = onFileError || window.alert;
-    const fileExtension = file.name.split(".").pop() || "";
-    const extensionFileError = checkExtension(fileExtension);
-    if (extensionFileError) {
-      return onFileUploadError(extensionFileError);
+    // Validate files
+    // If multiple files with image or multiple images
+    if (
+      files.length > 1 &&
+      files.some((file) => file.type.startsWith("image/"))
+    ) {
+      onFileUploadError("Multiple files with image are not supported");
+      return;
     }
 
-    if (isFileSizeExceeded(file)) {
-      return onFileUploadError(
-        `File size exceeded. Limit is ${fileSizeLimit / 1024 / 1024} MB`,
-      );
+    for (const file of files) {
+      const fileExtension = file.name.split(".").pop() || "";
+      const extensionFileError = checkExtension(fileExtension);
+      if (extensionFileError) {
+        onFileUploadError(extensionFileError);
+        continue;
+      }
+
+      if (isFileSizeExceeded(file)) {
+        onFileUploadError(
+          `File size exceeded. Limit is ${fileSizeLimit / 1024 / 1024} MB`,
+        );
+        continue;
```

Comment:
Do we need to process more files if there's an error?

Reply:
Updated. Great catch! I thought i committed from coderabbit's comment already.